### DISCUSSION TITLE
【keyball39】左手側の左右判定ができません

### DISCUSSION THREAD

**Discussion by hiroyamagen** [Yowkees/keyball#297] [Q&A]
お世話になります。
組み立て完了し、右手側(ball在り)は正常に動作しましたが、
左手側(ball無し)の動作に問題がありました。
キー入力はできるのですが、単体でPCに接続しても、右手側のキーマップでキー入力されてしまいます。
左右判定ジャンパーのはんだ付けは出来ていると思うのですが、確認方法、対応方法を教えていただけますか？
ジャンパーのはんだ付けは、何度かやり直しましたが、症状は変わりません。
ProMicroの接続を、ピンヘッダからコンスルーに変更した際、基板に負担がかかって接触が悪くなっているのかもしれません。

![DSC_0005](https://github.com/Yowkees/keyball/assets/16742030/a93af6a1-36f9-469b-afcd-b36b1e580984)

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。

LEFTジャンパがきちんと半田付けされていれば、下記画像の緑点どうしが導通しているはずです。テスターで導通確認可能でしょうか？
テスターが使用できない場合は、思い切って緑点どうしをエナメル線やゴム被覆導線等でショートしてみてください。
また、緑点を付けたダイオードの極性と半田付け不良がないかもご確認ください。
以上、よろしくお願いいたします。

![kb39_leftfix](https://github.com/Yowkees/keyball/assets/84209772/6d22a3c7-99e8-419c-abc8-0fb489c7c8a2)

**Comment by hiroyamagen** [comment]
ありがとうございます。
指摘頂いた点を確認しましたが、解決しませんでした。
緑点どおしは導通していました。
ダイオードの極性もあっていて、念のためはんだし直しました。
緑点どおしのショートも試しましたが、左手側とは認識出来ませんでした。
その他のはんだ付け部分も一通り半田ごてを当てなおしたのですが、症状変わらずです。
申し訳ありませんが、もう少しアドバイスを頂けないでしょうか？
以上、よろしくお願いします。

**Comment by Yowkees** [comment]
了解しました。
左右判定のみに不具合が出るのは回路上考えにくいですが、キーはすべて入力できますか？
ProMicroの不良の可能性もあり、ProMicroを左右入れ替えてみるとどうでしょうか？
また念のためファームウェアを再度書き直してみてください。

以上、お手数ですがご確認お願いいたします。

**Comment by hiroyamagen** [comment]
何度も申し訳ありません。

ProMicroの左右入れ替え：変化無し
ファームウェア書き直し：変化無し
キー入力の確認：1キーのみ反応なし
左側接続時
![left](https://github.com/Yowkees/keyball/assets/16742030/5bfc32a9-ee6c-45e7-835e-d5dddd769484)
反応しないキーのキーソケット側で、キー押下時に導通することは確認しました。
キー右上あたりにあるダイオードの向き、導通も大丈夫そうです。

(念のため右側接続時も確認　こちらは全キー反応あり)
![right](https://github.com/Yowkees/keyball/assets/16742030/9fd9a338-e976-4fd6-b9cc-f14cad078e51)

なんとか使えるようにしたいので、引き続きよろしくお願いします。

**Comment by Yowkees** [comment]
なるほど。左手基板は、親指キーにひとつ反応していないものがありますね。これがミソです。
ということで、COL5という信号線がProMicroまで繋がっていないようです。
下記写真の赤点の端子をご確認ください。ミドル基板の赤点とProMicroの対応する端子が導通しているでしょうか？（12Pコンスルーがうまく接触しているかの確認です）
また、回路的に赤点とピンク点２点はすべて導通しているはずですので、もし導通していない個所がありましたらエナメル線等の導線で接続してください。
以上、お手数ですが
![KB39_COL5fix](https://github.com/Yowkees/keyball/assets/84209772/94fb8a98-1a03-43be-9c23-7889aaee2dd8)
ご確認お願いいたします。

**Comment by hiroyamagen** [comment]
赤点とピンク点が導通していなかったです。
ここを短絡したら、無事左手として認識するようになりました。
いろいろ確認いただきありがとうございます。

これで使える、と思ったのですが、今度は左右をつないでもUSBをつないだ側しかキー入力できない、という問題が残っていました。
(今までは片側だけで動作確認をしていました)
今度もProMicro端子とどこかを短絡しなければならないと思うのですが、どのあたりをチェックすればよいでしょうか？
左右のジャック間が導通していることは確認しました。
タイトルと内容が変わってしまうので、ふさわしくないようであれあご指摘ください。

以上、よろしくお願いします。

**Comment by hiroyamagen** [comment]
DATAを短絡したら通信できるようになったようです。
メンテナンスしにくいのと、不格好なのですが、このような対応でよかったのでしょうか？

![DSC_0006](https://github.com/Yowkees/keyball/assets/16742030/9fc9d378-add1-46bd-bd10-8700c4ec7e35)

**Comment by Yowkees** [comment]
ご対応ありがとうございます。
左右通信はDATA線１本で通信していますので、ご対応いただいた修正でOKです。ご自身で調べていただきありがとうございました。

導線は裏面ですので、普段使いではまったく見えませんし、ちょっと苦労した跡が見えるのも自作キーボードの醍醐味かなと思っています。
気になるようでしたら、半田付けやり直したい方向けに基板単体や電子パーツとのセット等で販売することもできますので、ご希望の場合には白銀ラボウェブショップのお問い合わせからご連絡ください。

以上、今後ともどうぞよろしくお願いいたします。
